tests/integration/hive_metastore/test_migrate.py

```diff
@@ -101,3 +102,44 @@ def test_migrate_external_table(ws, sql_backend, inventory_schema, make_catalog,
 
     target_tables = list(sql_backend.fetch(f"SHOW TABLES IN {dst_schema.full_name}"))
     assert len(target_tables) == 1
+
+
+@retried(on=[NotFound], timeout=timedelta(minutes=5))
+def test_revert_migrated_table(ws, sql_backend, inventory_schema, make_schema, make_table):
+    schema1 = make_schema(catalog_name="hive_metastore")
+    schema2 = make_schema(catalog_name="hive_metastore")
+    tables = [
+        make_table(schema_name=schema1.name),
+        make_table(schema_name=schema1.name),
+        make_table(schema_name=schema2.name),
+        make_table(schema_name=schema2.name),
+    ]
+
+    for table in tables[0:3]:
+        # apply "upgraded to" table property to the first three tables.
+        logger.info(f"Applying upgraded_to table property for {table.full_name}")
+        sql_backend.execute(f"ALTER TABLE {table.full_name} set TBLPROPERTIES('upgraded_to'='FAKE DEST')")
+
+    crawler = TablesCrawler(sql_backend, inventory_schema)
+    tm = TablesMigrate(crawler, ws, sql_backend)
+    tm.revert_migrated_tables(schema=schema1.name)
+
+    # Checking that two of the tables were reverted and one was left intact.
+    # The first two tables belong to schema 1 and should have not "upgraded_to" property
+    assert not tm.checks_upgraded_to(tables[0].schema_name, tables[0].name)
```

Comment:
Don't rely on array order - someone will have to add/remove items to tables array and these asserts will be difficult to change

Reply:
fixed